packages/holocron-dev-server/README.md

```diff
@@ -0,0 +1,166 @@
+- local Holocron modules bundles analyzed and reported using [webpack bundle analyzer][webpack bundle analyzer]
+
+### Caveats
+- No support for custom `babel` config or auto-merge option (with zero-config)
+
+## 🤹‍ Usage
+
+> **Prerequisites**
+> - [Docker][docker] is required to load One App statics from development environment.
+> - [React Dev tools browser extension][React Dev tools extension] is needed in your browser for fast refresh to work
+
+### Installation
+
+```bash
+npm install -D @americanexpress/holocron-dev-server
+
+# or
+
+yarn add -D @americanexpress/holocron-dev-server
+```
+
+### Start Up
+
+add the script in the `package.json` of your Holocron module:
+
+```json
+{
+  "scripts": {
+    "dev": "holocron-dev-server"
+  }
+}
+```
+
+then run to start the development server:
+
+```bash
+npm run dev
+
+# or
+
+yarn dev
+```
+
+Once the command runs, the sandbox Holocron module reload server
+will start up using the configuration found in the modules `package.json`.
+
+### **Zero-Config**
+
+`holocron-dev-server` uses existing configurations from `"one-amex"`
+to source what modules, externals and other configuration like
+`rootModuleName` are used.
+
+The config uses [`one-amex.runner`][one-app-runner] (`modules`, `rootModuleName`, `moduleMapUrl`, `dockerImage`, `envVars`)
+and [`one-amex.bundler`][one-app-bundler] (`providedExternals`, `requiredExternals`)
+to set up the HMR environment for your Holocron module.
+
+#### **Parrot Scenarios**
+If a `mock/scenarios.js` exists in your module,
+the `parrot` scenarios will be watched for changes and updated,
+please note that the parrot browser extension will need to be changed to refer
+to the port used by the holocron development server.
+
+#### **Language Packs**
+The same applies if `locale/*` folder exists in your Holocron module.
+When a given locale gets modified, the holocron dev server will notify the client
+and will load the language pack into state.
+
+## 🎛️ API
+
+**The API is subject to change**
+
+### `holocron-dev-server`
+
+Command line tool usage.
+#### Usage
+
+```bash
+npx -p @americanexpress/holocron-dev-server -- holocron-dev-server
+```
+
+#### Configuration
+##### **`hmr` Options**
+
+Use the `one-amex.hmr` config for
+more advanced configurations. The experimental `hmr` config is
+subject to change. The options supported are:
+
+| name | type | required | value |
+|---|---|---|---|
+| `logLevel` | `String` | `false` | `info` or `4`, `log` or `3`, `warn` or `2`, `error` or `1`, `0` for silent |
+| `port` | `Number` | `false` |the port that the holocron dev server binds to |
+| `openWhenReady` | `Boolean` | `false` | Set to `true` to enable opening your default browser when ready |
+
+```json
+{
+  "one-amex": {
+    "hmr": {
+      "logLevel": "info",
+      "port": 4000,
+      "openWhenReady": false,
```

Comment:
This is awesome! did we consider making it default to true? and also adding a default url to open i.e, `localhost:4000/demo/my-module-name`

Reply:
this would be amazing, the only challenge we are facing is that the url most likely wouldn't be what the module is set to open to, we can add this in subsequent PRs to capture the ability to open a default set url just as you've mentioned.

Reply:
I really like the idea of the starting url, that would open up right to where a user wants to be!

I'd be hesitant to default opening the browser for two reasons: 1. some people may not like a browser opening for whatever reason, 2. it can be sporadic and open a browser randomly across desktops (last focused Chrome window is usually where it opens to) right when the dev server finishes its first webpack build and prints out important links to follow. I feel keeping it as an opt-in gives the logs more focus rather than jumping in to the experience, which may not be desired by unsuspecting users.